### PR TITLE
Data tiling: add tile sizes for riscv32 ukernel

### PR DESCRIPTION
**Pull request by lundong**
Created a new function enumerateMatmulTileRiscv32 to define tile sizes for riscv32 ukernel case.

### CODE REVIEW

**Comment by lundong** [comment]
@bjacob Hi Benoit, this is the PR per our discussions. Can you please help review the change? (not sure how to add you as a reviewer for now).

**Review by bjacob — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp [hunk 1]
```diff
@@ -53,6 +53,23 @@ enumerateMatmulTilesVMVX(EncodingUser user, ExecutableTargetAttr target) {
...
+// For narrow-{M,N} cases, this only enumerates on narrow M. The narrow-N cases
+// are handled by transposition in chooseMatmulTile.
+static SmallVector<TileMxNxK>
+enumerateMatmulTileRiscv32(EncodingUser user, ExecutableTargetAttr target) {
+  if (hasUkernel(target)) {
+    return {
+        TileMxNxK{8, 8, 4}, // Some reasonable tile shape.
+        TileMxNxK{4, 8, 4}, // Truncation of the above.
+        TileMxNxK{2, 8, 4}, // Truncation of the above.
+        TileMxNxK{1, 8, 4}, // Truncation of the above.
```
Comment (bjacob) [inline compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp:66]: There should be a if() on element types here - how can you know which tile size to use without checking element types?

**Review by bjacob — CHANGES_REQUESTED** [review]

File: compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp (hunk 1 above)
Comment (bjacob) [reply compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp:66]: oh, or are these just dummy sizes not particularly good for any particular element type? As a placeholder only for now? Then that's worth a comment.

Comment (lundong) [reply compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp:66]: Actually they are good choices of tile sizes for our use cases for all of our supported element types. And they also happen to be the same as the default VMVX ones. So shall I leave it as is?

Comment (bjacob) [reply compiler/src/iree/compiler/Codegen/Common/CPU/CPUMaterializeEncodingPass.cpp:66]: ok then.

**Review by bjacob — APPROVED** [review]

**Pull request merged**